jung-visualization/src/main/java/edu/uci/ics/jung/visualization/BaseVisualizationModel.java

```diff
@@ -130,6 +105,34 @@ public BaseVisualizationModel(
     }
   }
 
+  private LayoutModel<N, Point2D> createLayoutModel(Network<N, E> network, Dimension layoutSize) {
+    String modelStructure = System.getProperty("layout.model");
+    if (modelStructure == null) {
+      modelStructure = ModelStructure.QUAD_TREE.toString();
+    }
+    if (ModelStructure.GRID.toString().equals(modelStructure)) {
```

Comment:
can do this via a switch statement, I think that would be cleaner and clearer

(this applies whether it's a string or an enum)

Reply:
I really like that idea.